pkg/kubelet/status/status_manager.go

```diff
@@ -207,18 +232,78 @@ func (m *manager) Start() {
 	}, 0)
 }
 
-// GetPodResizeStatus returns the last cached ResizeStatus value.
-func (m *manager) GetPodResizeStatus(podUID types.UID) v1.PodResizeStatus {
+// GetPodResizeConditions returns the last cached ResizeStatus value.
+func (m *manager) GetPodResizeConditions(podUID types.UID) []*v1.PodCondition {
 	m.podStatusesLock.RLock()
 	defer m.podStatusesLock.RUnlock()
-	return m.podResizeStatuses[podUID]
+
+	if _, found := m.podResizeConditions[podUID]; !found {
+		return nil
+	}
+	return m.podResizeConditions[podUID].List()
+}
+
+// SetPodResizePendingCondition caches the last PodResizePending condition for the pod.
+func (m *manager) SetPodResizePendingCondition(podUID types.UID, reason, message string) {
+	m.podStatusesLock.Lock()
+	defer m.podStatusesLock.Unlock()
+
+	oldConditions, found := m.podResizeConditions[podUID]
+	if !found {
+		oldConditions = podResizeConditions{}
+	}
+
+	m.podResizeConditions[podUID] = podResizeConditions{
+		PodResizePending:    updatedPodResizeCondition(v1.PodResizePending, oldConditions.PodResizePending, reason, message),
+		PodResizeInProgress: oldConditions.PodResizeInProgress,
+	}
+}
+
+// SetPodResizeInProgressCondition caches the last PodResizeInProgress condition for the pod.
+func (m *manager) SetPodResizeInProgressCondition(podUID types.UID, reason, message string) {
```

Comment:
I don't think you need to handle it in this PR, but we should talk about how to surface errors in a follow-up PR. If the there's an error actuating the resize, we want to capture that in the condition, but it's not clear when that error condition should be cleared.

It might be sufficient to do something like: `if reason == "" { reason = oldCondition.reason; msg = oldCondition.msg }`